test/Azure.Functions.Cli.Tests/ZipHelperTests.cs

```diff
@@ -0,0 +1,197 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Functions.Cli.Common;
+using Azure.Functions.Cli.Helpers;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Azure.Functions.Cli.Tests
+{
+    public class ZipHelperTests
+    {
+        private readonly ITestOutputHelper _output;
+        private bool _isCI = Environment.GetEnvironmentVariable("TF_BUILD")?.ToLowerInvariant() == "true";
+
+        public ZipHelperTests(ITestOutputHelper output)
+        {
+            _output = output;
+
+            // reset to default in case other tests have set this up with mocks
+            FileSystemHelpers.Instance = null;
+        }
+
+        [Fact]
+        public async Task CreateZip_Succeeds()
+        {
+            var windowsZip = await BuildAndCopyFileToZipAsync("win-x64");
+            var linuxZip = await BuildAndCopyFileToZipAsync("linux-x64");
```

Comment:
can we include macos in these tests?

Reply:
good call -- let me sync with you on that

Reply:
discussed resolving this in a new PR